runtime/query.go

```diff
@@ -84,6 +94,11 @@ func populateFieldValueFromPath(msg proto.Message, fieldPath []string, values []
 		case reflect.Struct:
 			m = f
 			continue
+		case reflect.Map:
+			if !isLast {
+				return fmt.Errorf("unexpected map field in %s", strings.Join(fieldPath, "."))
```

Comment:
Can you elaborate in this error message. As I understand it the conern here is that you could make the querystring `?foo[bar]` which doesn't have a value. Could you add a remark about what was expected and why this is an error condition?

Reply:
This is to handle the case where an intermediate field in `fieldPath` refers to a map.  

i.e.  `obj.nested.map_field.foo` where `map_field` is a map.

Reply:
Could you enhance the error message to reflect that intent?

Reply:
sure no problem.